.github/scripts/cypress-test-finder.js

```diff
@@ -0,0 +1,257 @@
+ * @param {string} testName - Name of the test to run
+ * @param {number} burnIn - Number of times to run the test
+ * @param {string} ref - Git reference to use for the workflow
+ * @returns {string} - GitHub CLI command
+ */
+function generateWorkflowCommand(specPath, testName, burnIn = 20, ref = null) {
+  const refParam = ref ? `--ref ${ref}` : `--ref $(git branch --show-current)`;
+
+  return `gh workflow run e2e-stress-test-flake-fix.yml \
+  ${refParam} \
+  --field spec=${specPath} \
+  --field burn_in=${burnIn} \
+  --field grep="${testName}"`;
+}
+
+/**
+ * Main function
+ */
+function main() {
+  try {
+    // Parse command line arguments
+    const args = process.argv.slice(2);
+
+    // Get burn-in count
+    const burnIn = args.find((arg) => arg.startsWith("--burn="))
+      ? args.find((arg) => arg.startsWith("--burn=")).split("=")[1]
+      : 20;
+
+    // Check if we should execute the workflows
+    const shouldExecute = args.includes("--execute");
+
+    // Get base branch for comparison (default: master)
+    const baseBranchArg = args.find((arg) => arg.startsWith("--base="));
+    const baseBranch = baseBranchArg ? baseBranchArg.split("=")[1] : "master";
+
+    // Get git ref for workflow execution
+    const refArg = args.find((arg) => arg.startsWith("--ref="));
+    const ref = refArg ? refArg.split("=")[1] : null;
+
+    console.log(`Comparing changes against base branch: ${baseBranch}`);
+    if (ref) {
+      console.log(`Using head ref for workflows: ${ref}`);
+    }
+
+    // Get diff from base branch to HEAD
+    const diffOutput = execSync(`git diff ${baseBranch}...HEAD`).toString();
+
+    // Parse the diff to get changed files and lines
+    const changedFiles = parseDiff(diffOutput);
+
+    if (changedFiles.length === 0) {
+      console.log("No Cypress spec files changed");
+      return;
+    }
+
+    // Find affected tests in each changed file
+    let allAffectedTests = [];
+
+    for (const file of changedFiles) {
+      const tests = findAffectedTests(file.filePath, file.changedLines);
+      allAffectedTests = allAffectedTests.concat(tests);
+    }
+
+    // Output the results as workflow commands
+    if (allAffectedTests.length === 0) {
+      console.log("No affected tests found");
+    } else {
+      console.log(`Found ${allAffectedTests.length} affected tests\n`);
+
+      allAffectedTests.forEach((test) => {
+        const relativePath = path.relative(process.cwd(), test.file);
+        console.log(`Test: ${relativePath}: "${test.name}"`);
+
+        // Generate workflow command with the specific ref if provided
+        const command = generateWorkflowCommand(
+          relativePath,
+          test.name,
+          burnIn,
+          ref,
+        );
+        console.log(command);
+
+        if (shouldExecute) {
+          console.log(`\nExecuting workflow for "${test.name}"...`);
+          try {
+            execSync(command, { stdio: "inherit" });
```

Comment:
instead of executing it here, What about just triggering a workflow dispatch on the stress test workflow?

This could potentially take a billion years if you're stress testing a bunch of workflows on one runner. Even if we farm this out to separate runners via workflow dispatch, we should probably add a cap so we don't trigger 100 bonus jobs.